app/src/main/java/org/sagebionetworks/research/mpower/medication/MedicationAdapter.kt

```diff
@@ -0,0 +1,193 @@
+                            0 -> listener.enableNext(false)
+                            else -> listener.enableNext(true)
+                        }
+                    }
+
+                    override fun afterTextChanged(p0: Editable?) {}
+                })
+            }
+            SCHEDULE -> {
+                var sh = holder as ScheduleViewHolder
+                var schedule = item as Schedule
+                sh.bindView(schedule)
+                sh.checkbox.isChecked = schedule.anytime
+                sh.checkbox.setOnCheckedChangeListener{ buttonView, isChecked ->
+                    if(isChecked) {
+                        schedule.anytime = true
+                        listener.showAddSchedule(false)
+                        listener.setAnytime(schedule,true)
+                    } else {
+                        schedule.anytime = false
+                        listener.showAddSchedule(true)
+                        listener.setAnytime(schedule,false)
+                    }
+                }
+                if(schedule.anytime) {
+                    sh.dayContainer.visibility = View.GONE
+                    sh.timeContainer.visibility = View.GONE
+                } else {
+                    sh.dayContainer.visibility = View.VISIBLE
+                    sh.timeContainer.visibility = View.VISIBLE
+                }
+                sh.timeText.text = schedule.time
+                sh.timeContainer.setOnClickListener {
+                    view ->
+                        LOGGER.debug("Time clicked")
+                        getDate(context, sh.timeText, schedule)
+                }
+                sh.dayContainer.setOnClickListener {
+                    view ->
+                        LOGGER.debug("Day container clicked")
+                        listener.showDaySelection(name, schedule)
+                }
+                if(schedule.everday) {
+                    sh.dayText.text = "Everyday"
+                } else {
+                    sh.dayText.text = schedule.days.joinToString("," )
+                }
+            }
+
+            ADD -> {
+                var ah = holder as AddViewHolder
+                ah.bindView(item as Add)
+                ah.button.setOnClickListener {
+                    view ->
+                        LOGGER.debug("Add button clicked.")
+                        listener.addSchedule()
+                }
+            }
+
+        }
+
+    }
+
+    override fun getItemViewType(position: Int): Int {
+
+        val type = when (items[position].type) {
+            DOSAGE -> 0
+            SCHEDULE -> 1
+            ADD -> 2
+        }
+        return type
+    }
+
+    fun getDate(context: Context, textView: TextView, schedule: Schedule) {
+
+        var sdf = SimpleDateFormat("hh:mm aa")
+        val cal = Calendar.getInstance()
+        cal.setTime(sdf.parse(schedule.time))
+
+
+        val timeSetListener = TimePickerDialog.OnTimeSetListener { timePicker, hours, minutes ->
```

Comment:
I like this way of doing a TimePickerDialog a lot better than the TimePickerFragment I wrote. I'm going to refactor my code to use this approach.